gilbreth_perception/src/segmentation_node.cpp

```diff
@@ -0,0 +1,105 @@
+#include <pcl/console/parse.h>
+#include <pcl/filters/extract_indices.h>
+#include <pcl/filters/passthrough.h>
+#include <pcl/filters/voxel_grid.h>
+#include <pcl/kdtree/kdtree_flann.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+#include <pcl/segmentation/extract_clusters.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <ros/ros.h>
+#include <sensor_msgs/PointCloud2.h>
+
+// Algorithm params
+float down_sample(0.01);
+
+ros::Publisher pub;
+
+void loadParameter() {
+  // General parameters
+  std::map<std::string, float> parameter_map;
+  ros::NodeHandle ph("~");
+  ph.getParam("parameters", parameter_map);
+  down_sample = parameter_map["down_sample"];
+}
+
+void cloud_cb(const sensor_msgs::PointCloud2ConstPtr &cloud_msg, int argc, char **argv) {
```

Comment:
According to the naming rules of this function should be `cloudCb`

Reply:
Changed, I put too much attention on the recognition code, forgot to change the segmentation code.

Reply:
@lianjunli  were you able to push the change?

Reply:
@jrgnicho I am looking for how to make my kinetic-devel branch up to date, I think the original repo has changed because Michael's code was merged, I am searching how to make my fork sync with the original repo.

Reply:
@lianjunli  you can run **git rebase ...** as described below